File: Data Structure/1.Array/intersection_of_two_arrays_ii.py
# https://leetcode.com/problems/intersection-of-two-arrays-ii/
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        i = 0
        while(i<len(nums1)):
            if nums1[i] in nums2:
                nums2.remove(nums1[i])
                i+=1
            else:
                nums1.pop(i)
                
        return nums1
                
#         res = []
#         for i in nums1:
#             if i in nums2:
#                 res.append(i)
#                 nums2.remove(i)
        
#         return res

# --------------------------------------------------------------------

#         nums1.sort()
#         nums2.sort()
#         i=0
#         j=0
        
#         while(i<len(nums1) and j<len(nums2)):
#             if nums1[i] == nums2[j]:
#                 i+=1
#                 j+=1
                
#             elif nums1[i]<nums2[j]:
#                 nums1.pop(i)
#             else:
#                 j+=1
                
#         return nums1[:i]

# -------------------------------------------------------------------------
        
#         a = []
        
#         nums1.sort()
#         nums2.sort()
        
#         i = 0
#         j = 0
        
#         while(i<len(nums1) and j<len(nums2)):
            
#             if nums1[i] == nums2[j]:
#                 a.append(nums1[i])
#                 i+=1
#                 j+=1
                
#             elif nums1[i]<nums2[j]:
#                 i+=1
#             else:
#                 j+=1
                
#         return a

File: Daily Coding Problem/Amazon/Intersection of Two Arrays II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int cnt = 0;
        for (int i=0;i<nums1.length;i++){
            boolean ispresent = false;
            for(int j=0;j<nums2.length;j++){
                if(nums1[i] == nums2[j]){
                    nums2[j] = -1;
                    ispresent = true;
                    break;
                }
            }
            if(!ispresent){
              nums1[i] = -1;
                cnt+=1;
            }
        }
        int [] arr = new int[nums1.length-cnt];
        int k = 0;
        for(int i=0;i<nums1.length;i++){
            if(nums1[i] != -1){
                arr[k++] =  nums1[i];
                
            }
        }
        
        return  arr;
    }
}